src/Power.cpp

```diff
@@ -226,7 +226,46 @@ class AnalogBatteryLevel : public HasBatteryLevel
             digitalWrite(ADC_CTRL, LOW);
         }
 #endif
-#else  // ADC2
+#else // ADC2
+#ifdef ADC_CTRL
+#if defined(HELTEC_WIRELESS_PAPER) || defined(HELTEC_WIRELESS_PAPER_V1_0)
+        pinMode(ADC_CTRL, OUTPUT);
+        digitalWrite(ADC_CTRL, LOW); // ACTIVE LOW
+        delay(10);
+#endif
+#endif // End ADC_CTRL
+
+#ifdef CONFIG_IDF_TARGET_ESP32S3 // ESP32S3
+        // ADC2 wifi bug workaround not required, breaks compile
+        // On ESP32S3, ADC2 can take turns with Wifi (?)
+
+        const uint8_t max_attempts = 10;
+        int32_t adc_buf;
+        esp_err_t read_result;
+
+        // Multiple samples
+        for (int i = 0; i < BATTERY_SENSE_SAMPLES; i++) {
+            adc_buf = 0;
+            read_result = -1;
+
+            // Multiple attempts to read, incase WiFi blocks (very uncommon?)
+            for (int attempts = 0; attempts < max_attempts; attempts++) {
+                read_result = adc2_get_raw(adc_channel, ADC_WIDTH_BIT_12, &adc_buf);
+                if (read_result == ESP_OK)
+                    break;
+            }
+
+            // Handle the result
+            if (read_result == ESP_OK)
+                raw += adc_buf;
+            else {
+                // How to handle read failure?
```

Comment:
Just a heads up: if ten read attempts fail, the current behavior is to record 0V for the sample. 
Not sure if this should be handled differently.

For what it's worth, during brief testing (with WiFi connected), I don't remember seeing a single failed reading, let alone ten, so it may be an uncommon scenario.

Reply:
check my PR, i added a valid reading counter so the division is made only by the valid readings, this will solve the invalid readign issue even if not probable.

Reply:
Good idea! https://github.com/meshtastic/firmware/pull/3208/commits/05dd680dfa54473f0bbbda1c957c12d956d1802c